pkg/ui/src/views/statements/planView.tsx

```diff
@@ -16,115 +16,274 @@ import _ from "lodash";
 import React from "react";
 
 import { cockroach } from "src/js/protos";
-import { intersperse } from "src/util/intersperse";
 import IExplainTreePlanNode = cockroach.sql.IExplainTreePlanNode;
-import ExplainTreePlanNode_IAttr = cockroach.sql.ExplainTreePlanNode.IAttr;
+import IAttr = cockroach.sql.ExplainTreePlanNode.IAttr;
+import {ToolTipWrapper} from "oss/src/views/shared/components/toolTip";
 
-export function PlanView(props: {
-  title: string,
-  plan: IExplainTreePlanNode,
-}) {
-  if (!props.plan) {
-    return <div className="plan-view">
-      <h3>{props.title}</h3>
-      <p>No plan captured yet.</p>
-    </div>;
-  }
-  return <div className="plan-view">
-    <h3>{props.title}</h3>
-    <div className="plan-node">
-      <ul>
-        <li>
-          <PlanNode node={props.plan} />
-        </li>
-      </ul>
-    </div>
-  </div>;
+interface IPlanNode {
+  name: string;
+  attrs: IAttr[];
+  children: IPlanNode[][];
+}
+
+/* ************************* HELPER FUNCTIONS ************************* */
+
+function planNode(name: string, attrs: any, children: IPlanNode[][]): IPlanNode {
+  return {
+    "name": name,
+    "attrs": attrs,
+    "children": children,
+  };
 }
 
-interface PlanNodeProps {
-  node: IExplainTreePlanNode;
+// TODO(celia): explain.
+function flattenTree(treePlan: IExplainTreePlanNode): IPlanNode[] {
+  const flattenedPlan = [ planNode(
+    treePlan.name,
+    treePlan.attrs,
+    []),
```

Comment:
would indent as

```
[
  planNode(
    treePlan.name,
    treePlan.attrs,
    [],
  )
]
```

(We really need to install https://prettier.io/ in this project…)

Reply:
agreed